app/commands/submission/test_run/init.rb

```diff
@@ -3,7 +3,11 @@ class TestRun
     class Init
```

Comment:
This can now take a custom git_sha to run against, and also has a status for `head_run` which we use both here, and when the job is completed to know whether we're dealing with a normal run or a run against git head. 

I'm open to a better name than `head_run` if you have an opinion!

Reply:
Having thought about this more, I think I'll switch to having another type column here, rather than `head_run` as a flag. Maybe `test_run_type` with `[submission, published_solution_git_head]` or something. Thoughts welcome.